pypeit/spectrographs/keck_hires.py

```diff
@@ -94,201 +259,303 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
+        Returns:
+            :obj:`list`: List of tuples, where each tuple provides the 1-indexed
+            detector numbers that can be combined into a mosaic and processed by
+            ``PypeIt``.
         """
-        Define how metadata are derived from the spectrograph files.
+        return [(1,2,3)]
 
-        That is, this associates the ``PypeIt``-specific metadata keywords
-        with the instrument-specific header cards using :attr:`meta`.
+    @property
+    def default_mosaic(self):
+        return self.allowed_mosaics[0]
+
+
+    def get_detector_par(self, det, hdu=None):
         """
-        super().init_meta()
-        self.meta['decker'] = dict(ext=0, card='DECKNAME')
+        Return metadata for the selected detector.
 
+        Args:
+            det (:obj:`int`):
+                1-indexed detector number.
+            hdu (`astropy.io.fits.HDUList`_, optional):
+                The open fits file with the raw image of interest.  If not
+                provided, frame-dependent parameters are set to a default.
 
-def indexing(itt, postpix, det=None,xbin=None,ybin=None):
+        Returns:
+            :class:`~pypeit.images.detector_container.DetectorContainer`:
+            Object with the detector metadata.
+        """
+        # Binning
+        binning = '1,1' if hdu is None else self.get_meta_value(self.get_headarr(hdu), 'binning')
+
+        # Detector 1
+        # TODO -- Fill these in properly for HIRES
+        detector_dict1 = dict(
+            binning         = binning,
+            det             = 1,
+            dataext         = 1,
+            specaxis        = 0,
+            specflip        = False,
+            spatflip        = False,
+            platescale      = 0.135,
+            darkcurr        = 0.0,
+            saturation      = 65535.,
+            nonlinear       = 0.86,
+            mincounts       = -1e10,
+            numamplifiers   = 1,
+            # TODO THese are place holders at present. Getting these numbers right requires parsing headers to determine
+            # which readout mode is being used.
+            gain            = np.atleast_1d([1.]),
+            ronoise         = np.atleast_1d([3.]),
+            )
+
+        # Detector 2. HACK
+        detector_dict2 = detector_dict1.copy()
+        detector_dict2.update(dict(
+            det=2,
+            dataext=2,
+            gain=np.atleast_1d([1.]),
+            ronoise=np.atleast_1d([3.])
+        ))
+
+
+        # Detector 3,. HACK
+        detector_dict3 = detector_dict1.copy()
+        detector_dict3.update(dict(
+            det=3,
+            dataext=3,
+            gain=np.atleast_1d([1.]),
+            ronoise=np.atleast_1d([3.])
+        ))
+
+        # Instantiate
+        detector_dicts = [detector_dict1, detector_dict2, detector_dict3]
+        detector = detector_container.DetectorContainer(
+            **detector_dicts[det-1])
+
+        if hdu is None:
+            return detector
```

Comment:
Why do we need this?  `detector` is returned regardless of whether or not `hdu is None`.

Reply:
changed